packages/axios-extension/src/abap/adt-catalog/services/filestore-service.ts

```diff
@@ -0,0 +1,111 @@
+    }
+
+    /**
+     * If target `path` is a file, the file content is returned as string type.
+     * If target `path` is a folder, files and folders in this folder are returned as an array list
+     * of ArchiveFileNode objects.
+     *
+     * @see ArchiveFileNode
+     * @param type
+     *  Specifies if input `path` refers to a file or a folder. When starting exploring
+     *  the file structure from the root, type should be set to `folder`. The type information
+     *  of files and folders inside root folder can be found in the returned `ArchiveFileNode` entries.
+     * @param appName Deployed Fiori app name
+     * @param path
+     *   Default value is empty string. In this case the output would be folder structure information of the root folder.
+     *   Otherwise provide path to a folder or a file in the deployed archive. E.g. `/webapp/index.html`.
+     * @returns Folder content (ArchiveFileNode[]) | file content (string)
+     */
+    public async getAppArchiveContent<T extends ArchiveFileNodeType>(
+        type: T,
+        appName: string,
+        path = ''
+    ): Promise<ArchiveFileContentType<T>> {
+        const contentType = type === 'folder' ? 'application/atom+xml' : 'application/octet-stream';
+        const config = {
+            headers: {
+                Accept: 'application/xml',
+                'Content-Type': contentType
+            }
+        };
+        const encodedFullPath = encodeURIComponent(`${appName}${path}`);
+        const response = await this.get(`/${encodedFullPath}/content`, config);
+        return this.parseArchiveContentResponse(appName, response.data, type);
+    }
+
+    /**
+     * Parse response data from ADT service. If the content is XML document of
+     * folder structure, this method returns a list of `ArchiveFileNode` object. If the content
+     * is text string, this method returns the text cotent.
+     *
+     * @see ArchiveFileNode
+     * @param appName Deployed Fiori app name
+     * @param responseData Response from ADT service
+     * @param type Reponse data is the file content or folder content.
+     * @returns Folder content (ArchiveFileNode[]) | file content (string)
+     */
+    private parseArchiveContentResponse<T extends ArchiveFileNodeType>(
+        appName: string,
+        responseData: string,
+        type: T
+    ): ArchiveFileContentType<T> {
+        // File content that is not xml data.
+        if (type === 'file' || XmlParser.validate(responseData) !== true) {
+            return responseData as ArchiveFileContentType<T>;
+        }
+        // A list of file/folder items in the response data as xml string.
+        const options = {
+            attributeNamePrefix: '',
+            ignoreAttributes: false,
+            ignoreNameSpace: true,
+            parseAttributeValue: true
+        };
+        const obj = XmlParser.getTraversalObj(responseData, options);
+        const parsed = XmlParser.convertToJson(obj, options);
```

Comment:
Could either of these methods throw errors? If so, does it make sense to wrap the code in a try/catch block?

Reply:
XmlParser.validate(responseData) is supposed to be called first and guard the xml content parsing. Invalid xml has been covered by unit tests.